### PR TITLE
Support MySQL mathematical calculation Local Configuration

### PR DESCRIPTION
**Pull request by mengchuang123**
**What this PR does**:
Support MySQL mathematical calculation Local Configuration.
Add a configuration `enable_local_math_computation: True` to the file `bootstrap.yaml`, and we can calculate through arana , if `False`, calculate through Database. 
e.g:
`select 1+1, RAND(), ABS(-2.1)+1;`

**Which issue(s) this PR fixes**:

Fixes # https://github.com/arana-db/arana/issues/593

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/mysql/server.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ import (
 	"context"
 	"crypto/rand"
 	"fmt"
+	"github.com/arana-db/arana/pkg/config"
```
Comment (dongzl) [inline pkg/mysql/server.go:25]: format import

File: pkg/runtime/ast/select.go [hunk 2]
```diff
@@ -49,6 +49,26 @@ type SelectStatement struct {
...
+func (ss *SelectStatement) DisplayName() string {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (ss *SelectStatement) phantom() selectElementPhantom {
+	//TODO implement me
+	panic("implement me")
+}
+
```
Comment (dongzl) [inline pkg/runtime/ast/select.go:71]: why add those method?

File: pkg/runtime/optimize/calculate_visitor.go [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+)
+
+var _ ast.Visitor = (*CalculateVisitor)(nil)
+
+type CalculateVisitor struct {
+	ast.BaseVisitor
+	args []proto.Value
+}
+
+func NewXCalcualtor(args []proto.Value) *CalculateVisitor {
```
Comment (dongzl) [inline pkg/runtime/optimize/calculate_visitor.go:46]: what's the meaning `X`?

**Comment by dongzl** [comment]
please fix CI failure.

**Review by dongzl — APPROVED** [review]

File: pkg/runtime/optimize/dml/select.go [hunk 4]
```diff
@@ -51,6 +52,51 @@ func init() {
 
 func optimizeSelect(ctx context.Context, o *optimize.Optimizer) (proto.Plan, error) {
 	stmt := o.Stmt.(*ast.SelectStatement)
+	if uconfig.IsEnableLocalMathCompu(false) && stmt.From == nil {
```
Comment (jjeffcaii) [inline pkg/runtime/optimize/dml/select.go:55]: Should pass over `context.Context`.

Comment (jjeffcaii) [reply pkg/runtime/optimize/dml/select.go:55]: Use `len(from) < 1` instead.

**Comment by jjeffcaii** [comment]
IMPORTANT: no need to reimplement a calculator, you can use `extvalue.Compute(...)` directly.

File: pkg/runtime/optimize/dml/select.go [hunk 5]
```diff
@@ -19,6 +19,7 @@ package dml
 
 import (
 	"context"
+	"github.com/arana-db/arana/pkg/runtime/misc/extvalue"
```
Comment (jjeffcaii) [inline pkg/runtime/optimize/dml/select.go:22]: reformat imports

File: pkg/runtime/optimize/dml/select.go [hunk 6]
```diff
@@ -51,6 +52,62 @@ func init() {
...
+	enableLocalMathComputation := ctx.Value(proto.ContextKeyEnableLocalComputation{}).(bool)
+	if enableLocalMathComputation && len(stmt.From) == 0 {
+		isLocalFlag := true
+		var columnList []string
+		var valueList []proto.Value
+		for i := range stmt.Select {
+			switch stmt.Select[i].(type) {
+			case *ast.SelectElementExpr:
+				var nodeInner *ast.PredicateExpressionNode
+				calculateNode := stmt.Select[i].(*ast.SelectElementExpr).Expression()
```
Comment (jjeffcaii) [inline pkg/runtime/optimize/dml/select.go:64]: Declare a variable instead, for example:
```go
switch t := stmt.Select[i].(type) {
case *ast.SelectElementExpr:
      _ = t.Expression()
    // ...
}

```

File: pkg/runtime/optimize/shard_visitor.go [hunk 7]
```diff
@@ -177,7 +177,15 @@ func (sd *ShardVisitor) VisitNotExpression(node *ast.NotExpressionNode) (interfa
 func (sd *ShardVisitor) VisitPredicateExpression(node *ast.PredicateExpressionNode) (interface{}, error) {
 	return node.P.Accept(sd)
 }
-
+func (sd *ShardVisitor) VisitSelectElementExpr(node *ast.SelectElementExpr) (interface{}, error) {
+	switch node.Expression().(type) {
```
Comment (jjeffcaii) [inline pkg/runtime/optimize/shard_visitor.go:181]: Same as above.

**Review by jjeffcaii — COMMENTED** [review]
pls fix

File: pkg/runtime/optimize/dml/select.go (hunk 5 above)
Comment (mengchuang123) [reply pkg/runtime/optimize/dml/select.go]: Processed

File: pkg/runtime/optimize/dml/select.go (hunk 6 above)
Comment (mengchuang123) [reply pkg/runtime/optimize/dml/select.go]: Processed

File: pkg/runtime/optimize/shard_visitor.go (hunk 7 above)
Comment (mengchuang123) [reply pkg/runtime/optimize/shard_visitor.go]: Processed

**Comment by dongzl** [comment]
add ut

**Review by jjeffcaii — APPROVED** [review]

**Pull request merged**